vcs/gitcmd/repo.go

```diff
@@ -221,6 +235,30 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 	return branches, nil
 }
 
+func (r *Repository) branchesWithCommit(commitID string) (map[string]bool, error) {
+	cmd := exec.Command("git", "branch", "--contains", commitID)
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
```

Comment:
I find it very helpful to have descriptive errors, but I see that some other recent contributions don't use them. I still think it's worth it.

```
return nil, fmt.Errorf("exec %v in %s failed: %s (output follows)\n\n%s", cmd.Args, cmd.Dir, err, out)
```

Note that as @shurcooL has noted, this doesn't include stderr. For that reason, you can punt on this and someone can fix error handling all around later.